tests/pages/firefox/home.py

```diff
@@ -26,3 +34,18 @@ def fb_container_is_displayed(self):
     def browser_menu_list(self):
         el = self.find_element(*self._browser_menu_list_locator)
         return MenuList(self, root=el)
+
+    def init_promo(self):
+        assert not self.is_promo_displayed, 'Promo detail is not displayed'
+        # scroll down page to trigger promo to display
+        self.scroll_element_into_view(*self._page_promise_content_locator)
+        self.wait.until(expected.visibility_of_element_located(
+            self._sticky_promo_modal_content_locator))
```

Comment:
@alexgibson This function checks that it isn't visible to start and then scrolls down the page to an element (but not so far as in the footer) and then waits for the promo to become visible.

Reply:
I think the problem here is the test is waiting for *any* portion of the sticky promo to be visible in the viewport. At that point, it then tries to click the close button (which can still be off screen because the promo is animating, hence the error message).

Using the existing method of checking for the `is_displayed` class is probably going to be more consistent:

```
promo = self.find_element(*self._sticky_promo_modal_content_locator)
self.wait.until(lambda s: 'is-displayed' in promo.get_attribute('class'))
```

Reply:
Awesome. Revised all the tests to leverage this.